src/intervals/rounding.jl

```diff
@@ -35,26 +47,51 @@ macro round(T, expr1, expr2)
             Interval(prevfloat($expr1), nextfloat($expr2))
 
         else # mode == :narrow
-            Interval(@with_rounding($T, $expr1, RoundDown), @with_rounding($T, $expr2, RoundUp))
+            lo = @with_rounding($T, $expr1, RoundDown)
+            hi = @with_rounding($T, $expr2, RoundUp)
+            Interval(lo, hi)
         end
-
     end
 end
 
-@doc doc"""`@thin_round` possibly saves one operation compared to `@round`.""" ->
-macro thin_round(T, expr)
+macro controlled_round(T, expr1, expr2)
```

Comment:
The idea is to be sure that we round up and down only "one step"; their separation should be the local `eps`. The problem comes when the floating point is too close to the `RoundNearest` result, so they are *confussed* with that. In those cases, rounding up and down yields a difference which is two times the local `eps`. And you want to keep your results as tight as possible...

The above implementation is not the best regarding performance (six rounding operations and various changes in rounding mode); this could be improved (the trick we learned from Tucker), but once we are sure everything works correctly and as tight as possible!